### ISSUE TITLE
shellenv.sh: Failure when user cannot write "${HOMEBREW_PATHS_FILE}"

### ISSUE THREAD

**Issue by eeide** [Homebrew/brew#18301]
### `brew doctor` output

```shell
Your system is ready to brew.
```

### Verification

- [X] My "`brew doctor` output" above says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` twice and am still able to reproduce my issue.
- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### `brew config` output

```shell
HOMEBREW_VERSION: 4.3.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: c763732882720f463e84a79ebc607ba54b81ea5d
Last commit: 2 days ago
Core tap JSON: 10 Sep 17:33 UTC
Core cask tap JSON: 10 Sep 17:33 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.46.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```

### What were you trying to do (and why)?

After using an administrator account to upgrade my installation of Homebrew, I logged into my normal user account and tried to start a new `tcsh` shell (an iTerm2 tab running a shell).

### What happened (include all command output)?

The shell started, but it encountered an error running the output of:
```
set noglob
eval `/opt/homebrew/bin/brew shellenv`
unset noglob
```

The error was:
```
/opt/homebrew/Library/Homebrew/cmd/shellenv.sh: line 29: /opt/homebrew/etc/paths: Permission denied
```

The relevant logic in `shellenv.sh` tries to create `/opt/homebrew/etc/paths` if it does not exist. The error occurs on the `printf` line:
```
    if [[ ! -f "${HOMEBREW_PATHS_FILE}" ]]
    then
      printf '%s/bin\n%s/sbin\n' "${HOMEBREW_PREFIX}" "${HOMEBREW_PREFIX}" >"${HOMEBREW_PATHS_FILE}" 2>/dev/null
    fi
```

### What did you expect to happen?

I expected the shell to start without error.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
I believe that the underlying issue is that immediately after updating Homebrew, `/opt/homebrew/etc/paths` did not exist. (This is ${HOMEBREW_PATHS_FILE}.)

When I logged out of my admin account, logged into my user account, and started a shell, my user account did not have permission to create the file (by writing into the Homebrew directory), and thus the error.

The error *went away* after I logged back into my admin account and started a new shell under that account. The admin account was able to create the file, and thus when I subsequently went back to my user account and started a shell, the error did not occur.

In summary, the error will occur if the "${HOMEBREW_PATHS_FILE}" does not exist and the user running the `shellenv.sh` script does not have permission to create it.

## Steps to reproduce

* [Preparation.] Using an appropriate account, delete the "${HOMEBREW_PATHS_FILE}" if it exists.

* Switch users to an account that cannot write in the installed Homebrew tree.

* Start a `tcsh` shell and run the following command:

eval `/opt/homebrew/bin/brew shellenv`
```
```

**Comment by carlocab** [comment]
Thanks for the report; should be fixed by #18302.
